.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,46 @@
+name: Build & maybe upload PyPI package
+
+on:
+  push:
+    branches:
+    tags: ["*"]
+  pull_request:
```

Comment:
If we intend only to support the GH release approach, we should remove the other hooks:
```suggestion
```

Reply:
I've simplified the triggers.

We want this to run for:

* pushes: to ensure the build step still works for code changes (e.g. we didn't break pacakging) - it won't attempt to deploy. Peace of mind for only 40s of CI.
* PRs: ditto
* `workflow_dispatch`: it's always helpful to enable the UI button for manual triggers ahead of time, also won't attempt deploy
* release: the actual PyPI deploy